spring-xsuaa/src/main/java/com/sap/cloud/security/xsuaa/autoconfiguration/XsuaaResourceServerJwkAutoConfiguration.java

```diff
@@ -41,11 +42,13 @@
 	private final Logger logger = LoggerFactory.getLogger(getClass());
 
 	@Bean
-	@ConditionalOnBean(XsuaaServiceConfiguration.class)
+	@ConditionalOnBean({XsuaaServiceConfiguration.class, RestOperations.class })
 	@ConditionalOnWebApplication(type = Type.SERVLET)
 	@ConditionalOnMissingBean
-	public JwtDecoder xsuaaJwtDecoder(XsuaaServiceConfiguration xsuaaServiceConfiguration) {
+	public JwtDecoder xsuaaJwtDecoder(XsuaaServiceConfiguration xsuaaServiceConfiguration, RestOperations restOperations) {
```

Comment:
AutoConfiguration now only works if you already have RestOperations, it would be nice to also Conditionally create RestOperations if they are not there already. We had issues with this when upgrading to 2.2

Reply:
Please help me understand this correctly: does the new AutoConfiguration breaks because you don't actually have RestOperations on the classpath?

Reply:
I have `RestOperations` on the classpath. I have/had a `@SpringTest` which manually lists classes to use. In the past it was enough to have `XsuaaResourceServerJwkAutoConfiguration` configured so that it was able to make a `JwtDecoder`, now however it can only make a JwtDecoder if you also have a `RestOperations` in your `ApplicationContext` (or Spring has a way to make one) 
When running with full Autoconfiguration spring-boot is able to make `RestOperations` so maybe it's a non issue. It was an unexpected change on my side though.